Error using mex

Hello, 
I face one issue to use Matlab bindings with following error statement. I am using Matlab2020b in Ubuntu system.
The following is the error when I execute the compile_matlab_bindings_for_precice.m.
Could I get some suggestions to solve this problem?

building with 'g++'.
Error using mex
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp: In member function
‘virtual void MexFunction::operator()(matlab::mex::ArgumentList, matlab::mex::ArgumentList)’:
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:176:70: warning: ‘bool
precice::SolverInterface::hasToEvaluateSurrogateModel() const’ is deprecated: The manifold mapping feature is no longer
supported. [-Wdeprecated-declarations]
bool result = interface->hasToEvaluateSurrogateModel();
^
In file included from
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:6:0:
/usr/include/precice/SolverInterface.hpp:260:79: note: declared here
[[deprecated("The manifold mapping feature is no longer supported.")]] bool hasToEvaluateSurrogateModel() const;
^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:182:65: warning: ‘bool
precice::SolverInterface::hasToEvaluateFineModel() const’ is deprecated: The manifold mapping feature is no longer supported.
[-Wdeprecated-declarations]
bool result = interface->hasToEvaluateFineModel();
^
In file included from
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:6:0:
/usr/include/precice/SolverInterface.hpp:275:79: note: declared here
[[deprecated("The manifold mapping feature is no longer supported.")]] bool hasToEvaluateFineModel() const;
^~~~~~~~~~~~~~~~~~~~~~
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:217:65: warning:
‘std::set precice::SolverInterface::getMeshIDs() const’ is deprecated: Use getMeshID() for specific mesh names instead.
[-Wdeprecated-declarations]
const std::set ids = interface->getMeshIDs();
^
In file included from
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:6:0:
/usr/include/precice/SolverInterface.hpp:350:84: note: declared here
[[deprecated("Use getMeshID() for specific mesh names instead.")]] std::set getMeshIDs() const;
^~~~~~~~~~
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:247100 warning:
narrowing conversion of ‘matlab::data::TypedArray::operator const [with T = int; size_t = long unsigned
int](https://github.com/precice/precice/issues/0)[](https://github.com/precice/precice/issues/size_t).matlab::data::ArrayElementTypedRef<int, true>::operator matlab::data::ArrayElementTypedRef<int, true>::reference()’
from ‘matlab::data::ArrayElementTypedRef<int, true>::reference {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
outputs[0] = factory.createArrayFromBuffer<int32_t>({1,size[0]}, std::move(ids_ptr));
^
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:258:105: warning:
narrowing conversion of ‘matlab::data::TypedArray::operator const [with T = int; size_t = long unsigned
int](https://github.com/precice/precice/issues/0)[](https://github.com/precice/precice/issues/size_t).matlab::data::ArrayElementTypedRef<int, true>::operator matlab::data::ArrayElementTypedRef<int, true>::reference()’
from ‘matlab::data::ArrayElementTypedRef<int, true>::reference {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
outputs[0] = factory.createArrayFromBuffer({1,size[0]}, std::move(positions_ptr));
^
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:269100 warning:
narrowing conversion of ‘matlab::data::TypedArray::operator const [with T = int; size_t = long unsigned
int](https://github.com/precice/precice/issues/0)[](https://github.com/precice/precice/issues/size_t).matlab::data::ArrayElementTypedRef<int, true>::operator matlab::data::ArrayElementTypedRef<int, true>::reference()’
from ‘matlab::data::ArrayElementTypedRef<int, true>::reference {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
outputs[0] = factory.createArrayFromBuffer<int32_t>({1,size[0]}, std::move(ids_ptr));
^
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:335:105: warning:
narrowing conversion of ‘matlab::data::TypedArray::operator const [with T = int; size_t = long unsigned
int](https://github.com/precice/precice/issues/0)[](https://github.com/precice/precice/issues/size_t).matlab::data::ArrayElementTypedRef<int, true>::operator matlab::data::ArrayElementTypedRef<int, true>::reference()’
from ‘matlab::data::ArrayElementTypedRef<int, true>::reference {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
outputs[0] = factory.createArrayFromBuffer({1,size[0]}, std::move(positions_ptr));
^
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:336:96: error: no
matching function for call to ‘matlab::data::ArrayFactory::createArrayFromBuffer<int32_t>(,
std::remove_reference<int*&>::type)’
outputs[1] = factory.createArrayFromBuffer<int32_t>({1,size[0]}, std::move(ids));
^
In file included from /home/u0148289/software/matlab/R2020b/extern/include/MatlabDataArray.hpp:22:0,
from /home/u0148289/software/matlab/R2020b/extern/include/mexAdapter.hpp:10,
from /home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:3:
/home/u0148289/software/matlab/R2020b/extern/include/MatlabDataArray/ArrayFactory.hpp:556:19: note: candidate: template matlab::data::TypedArray matlab::data::ArrayFactory::createArrayFromBuffer(matlab::data::ArrayDimensions,
matlab::data::buffer_ptr_t, matlab::data::MemoryLayout)
TypedArray createArrayFromBuffer(ArrayDimensions dims,
^~~~~~~~~~~~~~~~~~~~~
/home/u0148289/software/matlab/R2020b/extern/include/MatlabDataArray/ArrayFactory.hpp:556:19: note: template argument
deduction/substitution failed:
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:336:91: note: cannot
convert ‘std::move<int*&>(ids)’ (type ‘std::remove_reference<int*&>::type {aka int*}’) to type
‘matlab::data::buffer_ptr_t {aka std::unique_ptr<int [], void ()(void)>}’
outputs[1] = factory.createArrayFromBuffer<int32_t>({1,size[0]}, std::move(ids));
~~~~~~~~~^~~~~
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:417:104: warning:
narrowing conversion of ‘dim’ from ‘int32_t {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
outputs[0] = factory.createArrayFromBuffer({dim,size[0]}, std::move(values_ptr));
^
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:417:104: warning:
narrowing conversion of ‘dim’ from ‘int32_t {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:417:104: warning:
narrowing conversion of ‘matlab::data::TypedArray::operator const [with T = int; size_t = long unsigned
int](https://github.com/precice/precice/issues/0).matlab::data::ArrayElementTypedRef<int, true>::operator matlab::data::ArrayElementTypedRef<int, true>::reference()’
from ‘matlab::data::ArrayElementTypedRef<int, true>::reference {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:428:97: warning:
narrowing conversion of ‘dim’ from ‘int32_t {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
outputs[0] = factory.createArrayFromBuffer({dim,1}, std::move(value_ptr));
^
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:428:97: warning:
narrowing conversion of ‘dim’ from ‘int32_t {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:449:104: warning:
narrowing conversion of ‘sizeA’ from ‘int32_t {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
outputs[0] = factory.createArrayFromBuffer({sizeA,sizeB}, std::move(values_ptr));
^
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:449:104: warning:
narrowing conversion of ‘sizeA’ from ‘int32_t {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:449:104: warning:
narrowing conversion of ‘sizeB’ from ‘int32_t {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]
/home/u0148289/openfoam-adapter/matlab-bindings/+precice/@SolverInterface/private/preciceGateway.cpp:449:104: warning:
narrowing conversion of ‘sizeB’ from ‘int32_t {aka int}’ to ‘long unsigned int’ inside { } [-Wnarrowing]

Error in compile_matlab_bindings_for_precice (line 19)
mex(strcat(path_Interface,".cpp"),"-output",path_Interface,flags{:});

--- a/+precice/@SolverInterface/private/preciceGateway.cpp
+++ b/+precice/@SolverInterface/private/preciceGateway.cpp
@@ -333,7 +333,7 @@ public:
                 double* positions = positions_ptr.get();
                 interface->getMeshVertices(meshID[0],size[0],ids,positions);
                 outputs[0] = factory.createArrayFromBuffer<double>({1,size[0]}, std::move(positions_ptr));
-                outputs[1] = factory.createArrayFromBuffer<int32_t>({1,size[0]}, std::move(ids));
+                outputs[1] = factory.createArrayFromBuffer<int32_t>({1,size[0]}, std::move(ids_ptr));
                 break;
             } 
             case FunctionID::isMeshConnectivityRequired:

